src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy

```diff
@@ -113,4 +384,213 @@ class PartialUpdateSpec extends HealthCheckSpecification {
         cleanup: "Remove the flow"
         flowHelperV2.deleteFlow(flow.flowId)
     }
+
+    def "Partial update with empty body does not actually update flow in any way"() {
+        given: "A flow"
+        def swPair = topologyHelper.switchPairs.first()
+        def flow = flowHelperV2.randomFlow(swPair)
+        flowHelperV2.addFlow(flow)
+        def originalCookies = northbound.getSwitchRules(swPair.src.dpId).flowEntries.findAll {
+            Cookie.isIngressRulePassThrough(it.cookie) || !Cookie.isDefaultRule(it.cookie)
+        }*.cookie
+
+        when: "Request a flow partial update without specifying any fields"
+        def flowBeforeUpdate = northboundV2.getFlow(flow.flowId)
+        northboundV2.partialUpdate(flow.flowId, new FlowPatchV2())
+
+        then: "Flow is left intact"
+        expect northboundV2.getFlow(flow.flowId), sameBeanAs(flowBeforeUpdate)
+                .ignoring("lastUpdated")
+                .ignoring("diverseWith")
+
+        and: "Flow rules have not been reinstalled"
+        northbound.getSwitchRules(swPair.src.dpId).flowEntries*.cookie.containsAll(originalCookies)
+
+        cleanup: "Remove the flow"
+        flowHelperV2.deleteFlow(flow.flowId)
+    }
+
+    @Tidy
+    @Unroll
+    def "Unable to partial update a flow in case new port is an isl port on a #data.switchType switch"() {
+        given: "An isl"
+        Isl isl = topology.islsForActiveSwitches.find { it.aswitch && it.dstSwitch }
+        assumeTrue("Unable to find required isl", isl as boolean)
+
+        and: "A flow"
+        def flow = flowHelperV2.randomFlow(isl.srcSwitch, isl.dstSwitch)
+        flowHelperV2.addFlow(flow)
+
+        when: "Try to edit port to isl port"
+        northboundV2.partialUpdate(flow.flowId, new FlowPatchV2().tap {
+            it."$data.switchType" = new FlowPatchEndpoint().tap { it.portNumber = isl."$data.port" }
+        })
+
+        then: "Error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
```

Comment:
in some test you use `error.statusCode` instead of `exc.rawStatusCode`
as for me would be great to use the same verification everywhere
for example in this case  `error.statusCode == HttpStatus.BAD_REQUEST`